src/Core.UnitTests/CSharpVB/RuleSetGeneratorTests.cs

```diff
@@ -234,6 +234,61 @@ public void RoslynRuleSet_Sonar_Rules_Added()
                 "Warning", "Warning", "None", "None");
         }
 
+        [TestMethod]
+        public void RoslynRuleSet_Rules_AreGroupAndSorted()
+        {
+            // Arrange
+            var generator = new RuleSetGenerator(new Dictionary<string, string>
+            {
+                { "sonaranalyzer-cs.analyzerId", "SonarAnalyzer.CSharp" },
+                { "sonaranalyzer-cs.ruleNamespace", "SonarAnalyzer.CSharp" },
+
+                { "wintellect.analyzerId", "AAA wintellect id" },
+                { "wintellect.ruleNamespace", "XXX wintellect namepace" },
+
+                { "myanalyzer.analyzerId", "ZZZ myanalyzer id" },
+                { "myanalyzer.ruleNamespace", "BBB myanalyzer namespace" }
+
+            });
+
+            var activeRules = new[]
+            {
+                CreateRule("roslyn.wintellect", "win2", true),
+                CreateRule("roslyn.wintellect", "win1", true),
+
+                CreateRule("roslyn.myanalyzer", "my 1", true),
+
+                CreateRule("csharpsquid", "S999", true),
+            };
+
+            var inactiveRules = new[]
+            {
+                CreateRule("roslyn.wintellect", "win0", false),
+                CreateRule("csharpsquid", "S111", false),
+            };
+
+            // Act
+            var ruleSet = generator.Generate("cs", activeRules, inactiveRules);
+
+            // Assert
+            ruleSet.Rules.Should().HaveCount(3);
+
+            // Expecting groups to be sorted alphabetically by analyzer id (not namespace)...
+            ruleSet.Rules[0].AnalyzerId.Should().Be("AAA wintellect id");
```

Comment:
It was an effort for me to understand the test, mostly because it was difficult to follow the line from setup to assertions. 
I think would've helped to have more descriptive names, such as   
`{ "first.analyzerId", "first analyzer id" }`
`ruleSet.Rules[0].AnalyzerId.Should().Be("first analyzer id");`

Reply:
With like `{ "first.analyzerId", "first analyzer id" }`, the test would still pass if the groups were sorted by the property key ("first.analyzer.id").
I've tried to simplify the setup slightly.

Reply:
(BTW if you've finished a review, please mark it as "requires changes" or "approved" so I know you're not still looking at it).

Reply:
Sorry my bad, I thought I marked it as Approved :)